cmd/pinniped-proxy/src/https.rs

```diff
@@ -33,6 +35,22 @@ fn validate_url(u: String) -> Result<String> {
     }
 }
 
+/// include_client_cert updates a tls connection to be built with a client cert for authentication.
+/// 
+/// The client cert is obtained by exchanging the authorization token for a client identity via
+/// pinniped.
+pub async fn include_client_identity_for_headers<'a>(mut tls_builder: &'a mut TlsConnectorBuilder, request_headers: HeaderMap<HeaderValue>, k8s_api_server_url: &str, k8s_api_ca_cert_data: &[u8]) -> Result<&'a mut TlsConnectorBuilder> {
```

Comment:
:astonished: this syntax is crazy

Reply:
Yeah, it's pretty unfamiliar at first, but necessary if you want to want the compiler to be able to determine the lifetimes of variables. It's usually not required to mention it explicitly, but is required here because there are two arguments being passed by reference, so we clearly identify that it's `tls_builder` which is required to have the same lifetime as the return value (because the return value references it). The book has a good overview: https://doc.rust-lang.org/book/ch10-03-lifetime-syntax.html